app/models/activity.rb

```diff
@@ -0,0 +1,43 @@
+class Activity < ActiveRecord::Base
+  belongs_to :object, polymorphic: true
+  belongs_to :actor, inverse_of: :activities, class_name: User.name
+  has_many :course_notifications, inverse_of: :activity, dependent: :destroy
+  has_many :user_notifications, inverse_of: :activity, dependent: :destroy
+
+  # Class methods used to create an activity.
+  class << self
+    # Create an activity.
+    #
+    # @param from the one triggers this activity.
+    # @param trackable the object that will be tracked in this activity.
+    # @param [Symbol] event the name of the activity indicating which event happened.
+    def create(from, trackable, event)
+      super({ actor: from, object: trackable, event: generate_key(trackable, event) })
```

Comment:
And actually you don't need to override this function. A `before_create` call back will do